src/Symfony/Component/DependencyInjection/Compiler/CheckServiceClassExistsPass.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Exception\IncorrectServiceClassException;
+use Symfony\Component\DependencyInjection\Reference;
+
+class CheckServiceClassExistsPass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container): void
+    {
+        foreach ($container->getDefinitions() as $id => $definition) {
+            if ($definition->isSynthetic()) {
+                continue;
+            }
+
+            if ($factory = $definition->getFactory()) {
+                $this->checkFactory($container, $factory, $id);
+                continue;
+            }
+
+            $class = $container->getParameterBag()->resolveValue($definition->getClass());
```

Comment:
not needed. In the èTYPE_AFTER_REMOVING` stage, classes have already been resolved in definitions in a persistent way.